src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+
+    public RightTriangle(String color, double firstLeg, double secondLeg) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    public double getFirstLeg() {
+        return firstLeg;
+    }
+
+    public double getSecondLeg() {
+        return secondLeg;
+    }
+
+    @Override
+    public double getArea() {
+        return 0.5 * firstLeg * secondLeg;
```

Comment:
The magic numbers '0.5' should be replaced with a named constant to improve readability and maintainability of the code. For example, you could define a constant named 'RIGHT_ANGLE_FACTOR' at the class level.